SwiftAppTemplate/SwiftAppTemplate/MainViewController.swift

```diff
@@ -0,0 +1,171 @@
+            connectReaderButton.trailingAnchor.constraint(equalTo: containerView.trailingAnchor, constant: -MainViewController.spacing),
+        ])
+
+        paymentTextField.translatesAutoresizingMaskIntoConstraints = false;
+        NSLayoutConstraint.activate([
+            paymentTextField.topAnchor.constraint(equalTo: connectReaderButton.bottomAnchor, constant: MainViewController.spacing),
+            paymentTextField.centerXAnchor.constraint(equalTo: containerView.centerXAnchor),
+            paymentTextField.leadingAnchor.constraint(equalTo: containerView.leadingAnchor, constant: MainViewController.spacing),
+            paymentTextField.trailingAnchor.constraint(equalTo: containerView.trailingAnchor, constant: -MainViewController.spacing),
+            paymentTextField.heightAnchor.constraint(equalTo: connectReaderButton.heightAnchor),
+        ])
+
+
+        collectPaymentButton.translatesAutoresizingMaskIntoConstraints = false;
+        NSLayoutConstraint.activate([
+            collectPaymentButton.topAnchor.constraint(equalTo: paymentTextField.bottomAnchor, constant: MainViewController.spacing),
+            collectPaymentButton.centerXAnchor.constraint(equalTo: containerView.centerXAnchor),
+            collectPaymentButton.bottomAnchor.constraint(equalTo: containerView.bottomAnchor),
+            collectPaymentButton.leadingAnchor.constraint(equalTo: containerView.leadingAnchor, constant: MainViewController.spacing),
+            collectPaymentButton.trailingAnchor.constraint(equalTo: containerView.trailingAnchor, constant: -MainViewController.spacing),
+        ])
+
+        loadingContainerView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            loadingContainerView.topAnchor.constraint(equalTo: view.topAnchor),
+            loadingContainerView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            loadingContainerView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            loadingContainerView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+        ])
+
+        loadingIndicator.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            loadingIndicator.centerXAnchor.constraint(equalTo: loadingContainerView.centerXAnchor),
+            loadingIndicator.centerYAnchor.constraint(equalTo: loadingContainerView.centerYAnchor),
+        ])
+
+        loadingLabel.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            loadingLabel.topAnchor.constraint(equalTo: loadingIndicator.bottomAnchor, constant: MainViewController.spacing),
+            loadingLabel.centerXAnchor.constraint(equalTo: loadingContainerView.centerXAnchor)
+        ])
+    }
+
+
+    @objc func discoverReader(sender: UIButton) {
+        loadingLabel.text = "Discovering Readers..."
```

Comment:
nit: maybe to make this more generic, we can drop the `loadingLabel` and just have the activity indicator?

```suggestion
```

Reply:
I think it may be worth adding, since when programing the collect payment portion which has mutliple steps you could update this and more clearly see where you are. I could remove this and just add a comment that this thing exists fill free to label what you are loading. Thoughts?

Reply:
Fair! Let's turn it into a comment - more fun for people to tweak
themselves!